src/main/java/edu/umich/its/cpm/GGBApiWrapper.java

```diff
@@ -68,202 +87,273 @@
 	}
 
-	// Take an URL and get the data back make calls to do_request to get
-	// external data.
+	//////////////////////////////////////////////////
+	// These <verb>_request methods deal with taking a request and dealing with issues
+	// that requires making multiple requests.  E.G. page links, throttling.
 
-	// This method with deal with any multiple call issues: e.g. authentication, page links,
-	// throttling.  It calls do_one_get_request to actually do the calls.
-
-	public JSONObject get_request(String url) {
-		// TODO: Multiple call request handing not yet implemented.
-		log.warn("Multiple call request handing not yet implemented.");
-		return do_one_get_request(create_complete_url(url));
+	public ApiResultWrapper get_request(String url) {
+		log.warn("get_request: Multiple call request handing not yet implemented.");
+		return runGet(create_complete_url(url));
 	}
 
-	// take a url, make a single call, return a json result.
-	protected JSONObject do_one_get_request(String url) {
-		log.debug("one_get_request: url: {}",url);
-		RestTemplate restTemplate = new RestTemplate();
-		ResponseEntity<String> value = restTemplate.getForEntity(url, String.class);
-		JSONObject valueObject = null;
-
-		if (value.getStatusCode().is2xxSuccessful()) {
-			valueObject = new JSONObject(value.getBody());
-		}
-
-		return valueObject;
+	// Do a post request
+	public ApiResultWrapper post_request(String url, String body) {
+		log.warn("post_result: Multiple call request handing not yet implemented.");
+		log.debug("post_request: {}", formatBodyInfo(url, body));
+		String full_url = create_complete_url(url);
+		return runPost(full_url, body);
 	}
 
-	// Make a partial url from a request into a full url based on the url prefix.
-	public String create_complete_url(String url) {
-		String completeUrl = baseUrl+url;
-		log.info("complete_url: {}",completeUrl);
-		return completeUrl;
+	public ApiResultWrapper put_request(String url, String body) {
+		log.warn("put_request: Multiple call request handing not yet implemented.");
+		log.debug("put_request: {}", formatBodyInfo(url, body));
+		ApiResultWrapper response = runPut(create_complete_url(url), body);
+		log.debug("put_request_response: {}", response.toString());
+		return response;
 	}
 
-	// Do a post request
-	public String post_request(String url, String body){
-		String full_url = create_complete_url(url);
-		HttpResponse response = runPost(full_url,body);
-		log.debug("post_request: {}",formatBodyInfo(url,body));
+	////////////////////////////////
+	// These run<VERB> methods take a URL, make a single call and return a result. 
+	
+	protected ApiResultWrapper runGet(String url) {
+		log.debug("one_get_request: url: {}", url);
+
+		HttpGet request = new HttpGet(url);
+		HttpResponse response = null;
 
-		if (response == null) {
-			// lower level should report the url and body
-			log.warn("post_response got null response");
-			return null;
+		try {
+			response = this.httpClient.execute(request);
```

Comment:
I like that you are just sending in the response who ever calling this needs to handle all the if else or case